sos/cleaner/mappings/ip_map.py

```diff
@@ -0,0 +1,189 @@
+# Copyright 2020 Red Hat, Inc. Jake Hunsaker <jhunsake@redhat.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+import ipaddress
+import random
+
+from sos.cleaner.mappings import SoSMap
+
+
+class SoSIPMap(SoSMap):
+    """A mapping store for IP addresses
+
+    Each IP address added to this map is chcked for subnet membership. If that
+    subnet already exists in the map, then IP addresses are deterministically
+    generated sequentially within that subnet. For example, if a given IP is
+    matched to subnet 192.168.1.0/24 then 192.168.1 may be obfuscated to
+    100.11.12.0/24. Each IP address in the original 192.168.1.0/24 subnet
+    will then be assigned an address in 100.11.12.0/24 sequentially, such as
+    100.11.12.1, 100.11.12.2, etc...
+
+
+    Internally, the ipaddress library is used to manipulate the address objects
+    however, when retrieved by SoSCleaner any values will be strings.
+    """
+
+    ignore_matches = [
+        '127.*',
+        '::1',
+        '0.0.0.0(.*)?',
+        '169.254.*',
+        '255.*'
```

Comment:
Got some false positives that are 0.*  

Thoughts on adding multicast addresses to the list too?   224.0.0.0/8–239.0.0.0/8 ?  Alternatively we could just stop filtering at 224+ as any above are either multicast or reserved for future use.

Reply:
What's the reasoning behind not filtering multicast? I'm sure users in secure environment would consider those network addresses still sensitive.

Reply:
True.. Maybe they should be masked to still be identifiable as multicast addresses?  That's my bigger concern, not being able to tell.

Reply:
That's certainly a reasonable concern. I'm not immediately sure how best to address it though - in no particular order here are some thoughts on it:

* We'd need to set a new range of obfuscated ranges.
    * Current design is addresses with a subnet (or in a known subnet) get assigned to a `100.x.y.z` address, with each new subnet incrementing the first octet. Address for which we aren't sure of the net get randomly generated, with each octet being between 1-99.

* We'd then need to prevent the non-multicast ranges from incrementing into the separate obfuscated range *and* into the actual multicast range. Granted, the latter is less of a concern since that would require 124+ discovered networks.

* Do we set the new obfuscated range above 239 (thus conflicting with other reserved ranges) or below 100 (meaning we'd need to restrict the random generation)?